### PR TITLE
(FM-4516) Add extensions to azure_vm

### PR DESCRIPTION
**Pull request by hunner**

### CODE REVIEW

**Comment by hunner** [comment]
Still needs docs.

**Comment by kenazk** [comment]
Docs PR can be found here:

https://github.com/puppetlabs/puppetlabs-azure/pull/88

**Comment by kenazk** [comment]
Closed the previous PR. Re-opened here: https://github.com/puppetlabs/puppetlabs-azure/pull/89

File: lib/puppet/provider/azure_vm/azure_arm.rb [hunk 1]
```diff
@@ -113,6 +130,35 @@ def create # rubocop:disable Metrics/AbcSize
...
       network_interface_name: default_based_on_name(resource[:network_interface_name]),
     })
+
+    self.extensions = resource[:extensions]
+
+  end
+
+  def extensions=(value)
+    value.each do |name, properties|
+      #properties: properties.split('_').map.with_index{ |x,i| i > 0 ? x.capitalize : x }.join,
```
Comment (bmjen) [inline lib/puppet/provider/azure_vm/azure_arm.rb]: This here for a future reference reason?

Comment (hunner) [inline lib/puppet/provider/azure_vm/azure_arm.rb]: It was, but I think I don't need it any more...

**Comment by DavidS** [comment]
@hunner this is failing on the (now fixed) puppet-lint issues. Can you please rebase this?

File: lib/puppet/provider/azure_vm/azure_arm.rb [hunk 2]
```diff
@@ -113,6 +130,34 @@ def create # rubocop:disable Metrics/AbcSize
...
+
+      if properties.is_a?(Hash)
+        create_extension({
+          :resource_group => resource[:resource_group],
+          :location       => resource[:location],
+          :vm_name        => resource[:name],
+          :name           => name,
+          :properties     => properties,
+        })
+      elsif properties == "absent"
```
Comment (DavidS) [inline lib/puppet/provider/azure_vm/azure_arm.rb]: This implementation does not match the docs, which say "Removing the extension parameter will delete the extension from the VM." One of them needs fixing, cc @kenazk

**Pull request merged**